Semester Title Customization + Adding Toolbox Placeholder Text

ccheung777 opened the pull request:
# Semester Title Customization
Users can now customize the title of their semesters. An edit icon has been added for a more intuitive interface, and a dashed line appears to indicate changes as well. Users can either click on the "Semester 1", "Semester 2", etc. where a dashed line appears to customize the placeholders or click on the edit icon to make changes. Additionally, when a user hovers over the title, directions such as "Click to Rename" and "Edit Semester Name" are added. 
Moreover, to match the changes in ```Toolbox``` placeholder text, the placeholder text for the semesters has changed in opacity and is italic. 
An example below: 
<img width="1085" height="416" alt="image" src="https://github.com/user-attachments/assets/94cf76d4-d73d-46db-988d-778118b9db99" />

# Toolbox Placeholder Text 
The ```Toolbox``` changed to contain placeholder messages, advising users to add courses to the toolbox to proceed. This improves user guidance and interface clarity. The logic implemented is to check whether there are any items in the ```Toolbox```. If there isn't, then we can display the placeholder message.
<img width="1676" height="127" alt="image" src="https://github.com/user-attachments/assets/b6a5125c-02c8-4585-a638-f58bc83cedbc" />

ccheung777 committed:
Added placeholder for Toolbox

Adds a message prompting users to add courses when the toolbox is empty, improving user guidance and interface clarity.

ccheung777 committed:
Added Custom Semester Titles

Added custom semester titles when adding a new semester and implemented inline editing of semester names in SemesterBlock. The UI now allows users to rename semesters directly, improving planner organization and usability.

ccheung777 committed:
Adjust SemesterBlock placeholder

Changed the placeholder text opacity from 60% to 50% in the SemesterBlock component for improved visual distinction.

ccheung777 committed:
Updates to Planner placeholder

Updated the PlannerCourseHolder component to include opacity-60 and italic styles for improved visual appearance.

ccheung777 committed:
PrettierWrite

Added trailing commas to callback functions in main.tsx and SemesterBlock.tsx to improve code consistency and formatting. No functional changes were made.

ccheung777 committed:
Removed opacity for Semester Title

Simplified the logic since we do not want the semester titles to have opacity.

ramonechen reviewed (changes requested):
Merge branch `main-preview` into this one and resolve any conflicts before it can be merged 🙏

ccheung777 committed:
Merge branch 'main-preview' into celine-branch

ccheung777 committed:
PrettierWrite

Removes unnecessary trailing comma in the useEffect setCount call for improved code clarity.

ccheung777 commented:
@ramonechen @mirmirmirr 
I fixed the merge conflict. It should be able to merge with ```main-preview``` now

ramonechen reviewed (approved):
I really like that semester names are truncated when past a certain length, depending on available screen size. Can't find any issues with this feature.
